src/components/Chat/Chat.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { PropsWithChildren } from 'react';
+
+import { useChat } from './hooks/useChat';
+
+import { ChatProvider } from '../../context/ChatContext';
+import { TranslationProvider } from '../../context/TranslationContext';
+
+import type { StreamChat } from 'stream-chat';
+import type { Streami18n } from '../../i18n/Streami18n';
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+  UnknownType,
+} from '../../../types/types';
+
+export type ChatProps<
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+> = {
+  /** The StreamChat client object */
+  client: StreamChat<At, Ch, Co, Ev, Me, Re, Us>;
+  /** Instance of Stream i18n */
+  i18nInstance?: Streami18n;
+  /** Initial status of mobile navigation */
+  initialNavOpen?: boolean;
+  /**
+   *
+   * Theme could be used for custom styling of the components.
+   *
+   * You can override the classes used in our components under parent theme class.
+   *
+   * e.g. If you want to build a theme where background of message is black
+   *
+   * ```
+   *  <Chat client={client} theme={demo}>
+   *    <Channel>
+   *      <MessageList />
+   *    </Channel>
+   *  </Chat>
+   * ```
+   *
+   * ```scss
+   *  .demo.str-chat {
+   *    .str-chat__message-simple {
+   *      &-text-inner {
+   *        background-color: black;
+   *      }
+   *    }
+   *  }
+   * ```
+   *
+   * Built in available themes:
+   *
+   *  - `messaging light`
+   *  - `messaging dark`
+   *  - `team light`
+   *  - `team dark`
+   *  - `commerce light`
+   *  - `commerce dark`
+   *  - `livestream light`
+   *  - `livestream dark`
+   */
+  theme?: string;
+};
+
+/**
+ *
+ * Chat - Wrapper component for Chat. The needs to be placed around any other chat components.
+ * This Chat component provides the ChatContext to all other components.
+ *
+ * ChatContext provides the following values:
+ *
+ * - channel - the current active channel
+ * - client - the client connection
+ * - closeMobileNav - function to close mobile navigation
+ * - mutes - array of muted users
+ * - navOpen - boolean if navigation is open
+ * - openMobileNav - function to open mobile navigation
+ * - setActiveChannel - a function to set currently active channel
+ * - theme - current theme
+ *
+ * It also exposes the withChatContext HOC which you can use to consume the ChatContext
+ *
+ * @example ../../docs/Chat.md
```

Comment:
need to move the `Chat.md` file from this `docs` directory to the same level as this and change the import to this. it will potential require changes to import paths within that markdown too but that's how the styleguidist will require it
```suggestion
 * @example ./Chat.md
```